### ISSUE TITLE
Hovering over C string, IntelliSense shows garbage data

### ISSUE THREAD

**Issue by J-Vaughan** [microsoft/vscode-cpptools#1577]
## Issue

When hovering over a string of any length in a C language file, IntelliSense
shows garbage data. This resembles what happens to me if I read out of bounds
in a string.

## Reproduction

With the C/C++ IntelliSense extension enabled, hover over a string inside a
macro or function, and the tooltip will show the string followed by a number of
garbage characters.

### Example 1

```c
puts("WIP");
```

#### With extension enabled

![screenshot from 2018-02-17 09-39-30](https://user-images.githubusercontent.com/24994262/36342234-533f590c-13c9-11e8-8982-d07f3f11149a.png)

#### With extension disabled

![screenshot from 2018-02-17 09-39-14](https://user-images.githubusercontent.com/24994262/36342233-532b6b04-13c9-11e8-8bae-8261cc84c46f.png)

### Example 2

```c
fprintf(stderr, "Usage: clog [options] message text...\n");
```

#### With extension enabled

![screenshot from 2018-02-17 10-03-39](https://user-images.githubusercontent.com/24994262/36342324-9cc1c8e8-13ca-11e8-8274-c6de760048d7.png)

#### With extension disabled

![screenshot from 2018-02-17 10-03-51](https://user-images.githubusercontent.com/24994262/36342287-0da60610-13ca-11e8-9670-57969c2c874c.png)

#### Environment info

- Extension Name: cpptools
- Extension Version: 0.15.0
- OS Version: Linux x64 4.13.0-32-generic
- VSCode version: 1.20.1

#### Data from clipboard

```json 
{
	"activationTimes": {
		"startup": false,
		"codeLoadingTime": 117,
		"activateCallTime": 9,
		"activateResolvedTime": 657,
		"activationEvent": "onLanguage:c"
	}
}
```

**Comment by sean-mcmanus** [comment]
I'm not able to repro this. Do you have UTF-8 characters > 1 byte in your file?

**Comment by bobbrow** [comment]
I think I know what this is.  I'll send you a PR, Sean.

**Comment by J-Vaughan** [comment]
Opening the file today, without making any edits or anything, doesn't reproduce the behaviour. It's strange, selectively showing itself.

On another note, the file is encoded in UTF-8. I'm also using Fira Code with ligatures enabled, if that matters. Thanks for looking into it!

Edit: As soon as I added a line (`puts("WIP");`) into the file, all the strings in the buffer started showing the garbage again. Another buffer was unaffected, until I added `puts("WIP");`. The tooltip shows the string having 4 characters (includes \0?), even though it only has 3 letters. Maybe it struggles with short strings?

**Comment by J-Vaughan** [comment]
For exact replication, the file I used and my settings are [here](https://gist.github.com/J-Vaughan/776484e9798107e5c53215d0b8c7304c)

**Comment by sean-mcmanus** [comment]
Yeah, it looks like Bob has a fix for this.

**Comment by J-Vaughan** [comment]
Thanks! :+1:

**Comment by sean-mcmanus** [comment]
@bobbrow This isn't fixed, right? Where you going to fix the remaining cases for this release? I heard Griffin was able to repro it, but I haven't seen it yet.

**Comment by bobbrow** [comment]
No, it sounds like my PR didn't actually address this issue.

**Comment by J-Vaughan** [comment]
Any news on this?

**Comment by bobbrow** [comment]
This won't make it into the March update, but we'll investigate it for April.

**Comment by J-Vaughan** [comment]
Thanks for being so attentative to these things

**Comment by mikedld** [comment]
Have the same issue on Mac. No Unicode chars in [the file](/transmission/transmission/blob/2.92/libtransmission/web.c). Reproducibility is low, I have no idea what's causing it.

![vscode-cpptools-issues-1577](https://user-images.githubusercontent.com/298203/39466905-eac8f140-4d33-11e8-8a91-bba0eca8aecf.gif)

EDIT: Happens right after VSCode start, w/o modifying the buffer.

**Comment by sean-mcmanus** [comment]
This should be fixed with 0.17.0-insiders2 or later: https://github.com/Microsoft/vscode-cpptools/releases . The final release is planned for early next week.

**Comment by J-Vaughan** [comment]
Thanks 👍

**Comment by bobbrow** [comment]
This is fixed in 0.17.0.

**Comment by J-Vaughan** [comment]
Thanks. Top notch team :)
